File: CSCI-240/Students/Date.cpp
// Date.cpp
// Naeem Tai
/* contains date data */

#include <iostream>
#include <sstream>
#include <stdlib.h>
#include "Date.h"

using namespace std;

// empty constructor
// set default month, day, and year
Date::Date()
{
    month = 1;
    day = 1;
    year = 1900;
}

// copy constructor
Date::Date(const Date& original){
    month = original.month;
    day = original.day;
    year = original.year;
}

// custom constructor
// set month, day, and date as given
Date::Date(int m, int d, int y)
{
    month = m;
    day = d;
    year = y;
}


/* setters */

// set month of the date
void Date::setMonth(int m){
    month = m;
}

// set day of the date
void Date::setDay(int d){
    day = d;
}

// set year of the date
void Date::setYear(int y){
    year = y;
}


/* getters */

// get month of the date
int Date::getMonth(){
    return month;
}

// get day of the date
int Date::getDay(){
    return day;
}

// get month of the date
int Date::getYear(){
    return year;
}


/* additional methods */

// print date
void Date::printDate(){
    cout << month << "/" << day << "/" << year << endl;
}

// set date data by separating a string
void Date::setData(string piece){
    // separate date data by /
    string value;
    istringstream ss(piece);
    int countt = 1;
    // set date data
    while (getline(ss, value, '/')){
        if(countt == 1){
            month = atoi(value.c_str());
        }
        else if(countt == 2){
            day = atoi(value.c_str());
        }
        else if(countt == 3){
            year = atoi(value.c_str());
        }
        countt++;
    }
}

// return date as a string
string Date::getDate(){
    stringstream d;
    d << month << "/" << day << "/" << year;
    return d.str();
}

File: CSCI-240/MultiUserApp/src/Date.java
import java.io.Serializable;
import java.util.Calendar;
import java.util.TimeZone;

// Date.java
// contains information related to date
// Naeem Tai
// 4/20/2013

public class Date implements Serializable{
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	// declare variables
	private int month;
	private int day;
	private int year;
	private boolean visible;
	
	// get current month, day, and year
	Calendar localCalendar = Calendar.getInstance(TimeZone.getDefault());
    int currentMonth = localCalendar.get(Calendar.MONTH) + 1;
    int currentDayOfMonth = localCalendar.get(Calendar.DAY_OF_MONTH);
    int currentYear = localCalendar.get(Calendar.YEAR);
	
	// constructor with no parameters
	public Date(){
		this.month = currentMonth;
		this.day = currentDayOfMonth;
		this.year = currentYear;
		this.visible = false;
	}// end constructor
		
	// constructor with parameters
	public Date(int m, int d, int y, boolean v){
		this.month = m;
		this.day = d;
		this.year = y;
		this.visible = v;
	}// end constructor
	
	// return date to be printed
	public String printDate(){
		if (this.visible){
			return this.month + "/" + this.day + "/" + this.year;
		}
		else{
			return "Hidden";
		}
	}
	
	// getters
	
	// get month of the date
	public int getMonth(){
		return this.month;
	}
	
	// get day of the date
	public int getDay(){
		return this.day;
	}
	
	// get year of the date
	public int getYear(){
		return this.year;
	}
	
	// get if date is visible or not
	public boolean isVisible(){
		return this.visible;
	}
	
	
	// setters
	
	// set month of the date
	public void setMonth(int m){
		this.month = m;
	}
	
	// set day of the date
	public void setDay(int d){
		this.day = d;
	}
	
	// set year of the date
	public void setYear(int y){
		this.year = y;
	}
	
	// set visibility of the date
	public void setVisible(boolean v){
		this.visible = v;
	}
	
}// end class
